components/logins/ios/Logins/LoginsStorage.swift

```diff
@@ -10,6 +10,11 @@ fileprivate let queue = DispatchQueue(label: "com.mozilla.logins-storage")
 open class LoginsStorage {
     private var raw: UInt64 = 0
     let dbPath: String
+    private var interrupt_handle: LoginsInterruptHandle?
+    // It's not 100% clear to me that this is necessary, but without it
+    // we might have a data race between reading `interrupt_handle` in
+    // `interrupt()`, and writing it in `doDestroy` (or `doOpen`)
```

Comment:
If `interrupt` and `close` are called concurrently, right?

Reply:
Yes. (Well, there are a couple functions that bottom out in places where we write. `close` (or `lock`) is the main user facing one. Since this is an internal comment, I'm being specific)